core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java

```diff
@@ -11283,4 +11284,22 @@ private static RelNode toLogical(RelNode rel, RelBuilder builder) {
 
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedSql));
   }
+
+  @Test public void viewWithProjectRelTrait() {
```

Comment:
please check whether this TC tests the added code 
try to put check on the added trait and assert the same